assets/js/phoenix.js

```diff
@@ -875,13 +876,13 @@ export class Socket {
    *
    * @param {Function} callback
    */
-  onError(callback){ this.stateChangeCallbacks.error.push(callback) }
+  onError(callback, ref = this.ref){ this.stateChangeCallbacks.error.push({ [ref]: callback }) }
```

Comment:
we need to return the ref to the caller, and introduce the `sockert.off` function. We also need to generate a new unique ref if one is not passed, ie:

```suggestion
  onError(callback, ref = this.makeRef()){ 
    this.stateChangeCallbacks.error.push({ref, callback})
    return ref
  }

```